explorer/interpreter/resolve_unformed.h

```diff
@@ -5,11 +5,90 @@
 #ifndef CARBON_EXPLORER_INTERPRETER_RESOLVE_UNFORMED_H_
 #define CARBON_EXPLORER_INTERPRETER_RESOLVE_UNFORMED_H_
 
+#include <optional>
+
 #include "explorer/ast/ast.h"
 #include "explorer/common/nonnull.h"
 
 namespace Carbon {
 
+// Maps AST nodes to flow facts within a function.
+class FlowFacts {
+ public:
+  enum class ActionType {
+    // Used at `VariableDefinition` with initialization.
+    AddInit,
+    // Used at `VariableDefinition` without initialization.
+    AddUninit,
+    // Used at AST nodes that potentially initializes a variable.
+    Form,
+    // Used at AST nodes that uses a variable.
+    Check,
+    // Used in traversing children nodes without an acion to take.
+    None,
+  };
+  // Take action on flow facts based on `ActionType`.
+  auto TakeAction(Nonnull<const AstNode*> node, ActionType action,
+                  SourceLocation source_loc, const std::string& name)
+      -> ErrorOr<Success>;
+
+ private:
+  enum class FormedState {
+    MustBeFormed,
+    MayBeFormed,
+    Unformed,
+  };
+  // Aggregate information about a AstNode being analyzed.
+  struct Fact {
+    FormedState formed_state;
+  };
+
+  auto get_fact(Nonnull<const AstNode*> node) const
```

Comment:
We'd typically expect CapitalCasing for methods like this (GetFact, AddFact); they're not simple accessors/mutators as I think of them.

Reply:
Changed to CapitalCasing.